birdhouse/pavics-compose.sh

```diff
@@ -157,43 +95,36 @@ done
 export AUTODEPLOY_EXTRA_REPOS_AS_DOCKER_VOLUMES
 
 # we apply all the templates
-find ./config ./components ./optional-components $EXTRA_CONF_DIRS -name '*.template' |
+find $ALL_CONF_DIRS -name '*.template' |
   while read FILE
   do
     DEST=${FILE%.template}
     cat ${FILE} | envsubst "$VARS" | envsubst "$OPTIONAL_VARS" > ${DEST}
   done
 
 if [ x"$1" = x"up" ]; then
-  # this is external in docker-compose.yml so have to create here
-  # no error if already exist, just an error message
-  docker network create jupyterhub_network
-
-  # no error if already exist
-  # create externally so nothing will delete these data volume automatically
-  docker volume create jupyterhub_data_persistence  # jupyterhub db and cookie secret
-  docker volume create thredds_persistence  # logs, cache
-
-  if [ ! -f "$GEOSERVER_DATA_DIR/global.xml" ]; then
-    echo "fix GeoServer data dir permission on first run only, when data dir do not exist yet."
-    FIRST_RUN_ONLY=1 $COMPOSE_DIR/deployment/fix-geoserver-data-dir-perm
-  fi
-
-  for adir in ${EXTRA_CONF_DIRS}; do
+  for adir in $ALL_CONF_DIRS; do
     COMPONENT_PRE_COMPOSE_UP="$adir/pre-docker-compose-up"
     if [ -x "$COMPONENT_PRE_COMPOSE_UP" ]; then
       echo "executing '$COMPONENT_PRE_COMPOSE_UP'"
       sh -x "$COMPONENT_PRE_COMPOSE_UP"
     fi
   done
-
 fi
 
 COMPOSE_CONF_LIST="-f docker-compose.yml"
-for adir in ${EXTRA_CONF_DIRS}; do
-  if [ -f "$adir/docker-compose-extra.yml" ]; then
-    COMPOSE_CONF_LIST="${COMPOSE_CONF_LIST} -f $adir/docker-compose-extra.yml"
```

Comment:
I renamed them because there isn't really a main `docker-compose.yml` anymore so the idea of there being a "main" one and "extra" ones don't really apply anymore.

Your point about them being able to be run as a standalone is a good one though. I tried not to change the docker-compose files that much but I believe that right now some of them could potentially be run standalone and others can't. This will depend on whether they depend on other components to run. 

We could do the following:

- if a docker compose file _can_ be run as a standalone: name it `docker-compose.yml`
- if a docker compose file _cannot_ be run as a standalone: name it `docker-compose-extra.yml`

In every case though they'll actually need to be run through `pavics-compose.sh` so I don't know if we really care about this distinction that much anymore.

I'm happy to change the names back to `docker-compose-extra.yml` if you'd prefer.

Reply:
Ok so you basically renamed `docker-compose-extra.yml` to `docker-compose.yml` simply because the main `docker-compose.yml` is now empty.

In that case, please revert to `docker-compose-extra.yml` for this change to be backward compat with all the components from external repos. Otherwise, all the external repos will also have to do the rename.  I think CRIM has at least one external repo.  PCIC has one repo for each of their bird.  Ouranos has quite a bit of external repos.   This is a lot of rename to be done in sync with this PR.

From `docker-compose` perspective, the first `docker-compose.yml` in the list, even if it is empty, is the main one, and all subsequent are fragments.  All relative paths (ex: volume-mount paths) in the fragments are relative to the main `docker-compose.yml`.  So I think, in the current state, most of the `docker-compose-extra.yml` can not become standalone, unless they do not have any config volume-mount in.  That would be cool but I don't think we should put in the effort to make them standalone, for now.

Reply:
Ok I changed the name back to docker-compose-extra.yml everywhere